src/System.Runtime/tests/CompareToTests.netstandard1.7.cs

```diff
@@ -0,0 +1,165 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Xunit;
+
+namespace System.Tests
+{
+    public static partial class CompareToTests
+    {
+        [Fact]
+        public static void FromBoolean()
+        {
+            object testValue = false;
+            object testValue2 = true;
+            object wrongValue = "hello";
+            bool val1 = false;
+            bool val2 = true;
+            Assert.Equal(0, val1.CompareTo(testValue));
+            Assert.True(0 > val1.CompareTo(testValue2));
+            Assert.True(0 < val2.CompareTo(testValue));
+            Assert.Throws<ArgumentException>(() => val1.CompareTo(wrongValue));
+        }
+
+        [Fact]
+        public static void FromChar()
+        {
+            object testValue = 'c';
+            char val1 = 'c';
+            char val2 = 'a';
+            char val3 = 'd';
+            object wrongValue = "hello";
+            Assert.Equal(0, val1.CompareTo(testValue));
+            Assert.True(0 > val2.CompareTo(testValue));
+            Assert.True(0 < val3.CompareTo(testValue));
+            Assert.Throws<ArgumentException>(() => val1.CompareTo(wrongValue));
+        }
+
+        [Fact]
+        public static void FromDouble()
+        {
+            object testValue = 1.1;
+            double val1 = 1.1;
+            double val2 = double.MinValue;
+            double val3 = double.MaxValue;
+            object wrongValue = "hello";
+            Assert.Equal(0, val1.CompareTo(testValue));
+            Assert.True(0 > val2.CompareTo(testValue));
+            Assert.True(0 < val3.CompareTo(testValue));
+            Assert.Throws<ArgumentException>(() => val1.CompareTo(wrongValue));
+        }
+
+        [Fact]
+        public static void FromInt16()
+        {
+            object testValue = (Int16)0;
+            Int16 val1 = 0;
+            Int16 val2 = Int16.MinValue;
+            Int16 val3 = Int16.MaxValue;
+            object wrongValue = "hello";
+            Assert.Equal(0, val1.CompareTo(testValue));
+            Assert.True(0 > val2.CompareTo(testValue));
+            Assert.True(0 < val3.CompareTo(testValue));
+            Assert.Throws<ArgumentException>(() => val1.CompareTo(wrongValue));
+        }
+
+        [Fact]
+        public static void FromInt32()
+        {
+            object testValue = (Int32)0;
+            Int32 val1 = 0;
+            Int32 val2 = Int32.MinValue;
+            Int32 val3 = Int32.MaxValue;
+            object wrongValue = "hello";
+            Assert.Equal(0, val1.CompareTo(testValue));
+            Assert.True(0 > val2.CompareTo(testValue));
+            Assert.True(0 < val3.CompareTo(testValue));
+            Assert.Throws<ArgumentException>(() => val1.CompareTo(wrongValue));
+        }
+
+        [Fact]
+        public static void FromInt64()
```

Comment:
>FromInt64 [](start = 27, length = 9)

can you add a case or two for each of these that use a different numeric type. Eg., compare Int64 to Int32 or Byte should work. 
Also comparing a floating point type to a shorter floating point type or to an integral type should work